trunk/Admin/AdminQuickbarAdmin.php

```diff
@@ -0,0 +1,446 @@
+            if ( empty( $posts ) ) {
+                continue;
+            }
+            if ( !$postType->hierarchical ) {
+                $output .= '<div class="admin-quickbar-category">' . $catName . '</div>';
+            }
+
+            $output .= $this->getLoopPosts( $postType, $posts );
+        }
+
+        return $output;
+    }
+
+    /**
+     * Get rendered posts loop
+     *
+     * @param $postType
+     * @param $posts
+     * @return string
+     * @throws \ImagickException
+     */
+    public function getLoopPosts( $postType, $posts ) {
+        $output = '';
+        foreach ( $posts as $post ) {
+            $style = $this->getMarginStyle( $post, $postType, $lastParent, $margin );
+            $postTypeInfo = $this->getPostTypeInfo( $postType, $post );
+
+            $tpl = new Template( self::PartialDir . '/loop-posts.php', [
+                'post' => $post,
+                'postTypeInfo' => $postTypeInfo,
+                'style' => $style,
+                'thumb' => $this->getThumb( $post ),
+                'postTitle' => $this->getPostTitle( $post ),
+            ] );
+            $output .= $tpl->getRendered();
+        }
+
+        return $output;
+    }
+
+    /**
+     * Returns post-thumb html
+     *
+     * @param \WP_Post $post
+     * @return string
+     * @throws \ImagickException
+     */
+    public function getThumb( $post ) {
+        $output = '';
+        if ( has_post_thumbnail( $post ) ) {
+            // from post-thumbnail
+            $attachmentId = get_post_thumbnail_id( $post->ID );
+            $path = get_attached_file( $attachmentId );
+            $url = wp_get_attachment_image_src( $attachmentId, 'thumbnail' );
+            $url = !empty( $url ) ? $url[0] : '';
+        } else if ( $post->post_type == 'attachment' ) {
+            // direct from attachment
+            $path = get_attached_file( $post->ID );
+            $url = wp_get_attachment_image_src( $post->ID, 'thumbnail' );
+            $url = !empty( $url ) ? $url[0] : '';
+        }
+
+        if ( empty( $url ) && class_exists( 'Lib\PostGalleryImageList' ) ) {
+            // from post-gallery
+            $postGalleryImages = \Lib\PostGalleryImageList::get( $post->ID );
+            if ( count( $postGalleryImages ) ) {
+                $firstThumb = array_shift( $postGalleryImages );
+                $path = $firstThumb['path'];
+            }
+        }
+
+        if ( !empty( $path ) && class_exists( 'Lib\Thumb' ) ) {
+            $path = explode( '/wp-content/', $path );
+            $path = '/wp-content/' . array_pop( $path );
+
+            $thumbInstance = new \Lib\Thumb();
+            $thumb = $thumbInstance->getThumb( array(
+                'path' => $path,
+                'width' => '150',
+                'height' => '150',
+                'scale' => '0',
+            ) );
+
+            if ( !empty( $thumb['url'] ) ) {
+                $output = '<img src="" data-src="'
```

Comment:
HTML should only be used in template files.

Reply:
Outsourced to template